### PR TITLE
Allow to set the BLE GAP name.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/ble/ble.toit [hunk 1]
```diff
@@ -252,9 +252,14 @@ class Adapter extends Resource_:
     reconnects more efficient.
 
   If $secure-connections is true then the peripheral is enabling secure connections.
+
+  If $name is provided it is used for the GAP name. The GAP name can be different from the
```
Comment (kasperl) [inline lib/ble/ble.toit:256]:
```suggestion
  If $name is provided, it is used for the GAP name. The GAP name can be different from the
```

File: lib/ble/ble.toit [hunk 2]
```diff
@@ -252,9 +252,14 @@ class Adapter extends Resource_:
...
+    advertised name in the advertisement data. On some platforms the GAP name is stored
```
Comment (kasperl) [inline lib/ble/ble.toit:257]:
```suggestion
    advertised name in the advertisement data. On some platforms, the GAP name is stored
```

File: src/compiler/propagation/type_primitive_ble.cc [hunk 3]
```diff
@@ -61,6 +61,7 @@ TYPE_PRIMITIVE_ANY(get_bonded_peers)
 TYPE_PRIMITIVE_ANY(toit_callback_init)
 TYPE_PRIMITIVE_ANY(toit_callback_deinit)
 TYPE_PRIMITIVE_ANY(toit_callback_reply)
+TYPE_PRIMITIVE_ANY(set_gap_device_name)
```
Comment (kasperl) [inline src/compiler/propagation/type_primitive_ble.cc:64]: Always returns null.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/ble/ble.toit (hunk 1 above)
Comment (floitsch) [reply lib/ble/ble.toit]: done.

File: lib/ble/ble.toit (hunk 2 above)
Comment (floitsch) [reply lib/ble/ble.toit]: done.

File: src/compiler/propagation/type_primitive_ble.cc (hunk 3 above)
Comment (floitsch) [reply src/compiler/propagation/type_primitive_ble.cc]: done.

**Pull request merged**